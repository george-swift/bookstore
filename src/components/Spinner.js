import { FaSpinner } from 'react-icons/fa';
import PropTypes from 'prop-types';

const Spinner = ({ loading }) => (
  loading && (
    <span>
      <FaSpinner className="loading" />
    </span>
  )
);

Spinner.propTypes = PropTypes.bool.isRequired;

export default Spinner;
