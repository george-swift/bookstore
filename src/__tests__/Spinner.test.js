import renderer from 'react-test-renderer';
import Spinner from '../components/Spinner';

describe('Testing Spinner component', () => {
  it('should render correctly', () => {
    const tree = renderer.create(<Spinner loading />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
