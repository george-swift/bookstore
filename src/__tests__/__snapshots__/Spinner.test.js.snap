// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Spinner component should render correctly 1`] = `
<span>
  <svg
    className="loading"
    fill="currentColor"
    height="1em"
    stroke="currentColor"
    strokeWidth="0"
    style={
      Object {
        "color": undefined,
      }
    }
    viewBox="0 0 512 512"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M304 48c0 26.51-21.49 48-48 48s-48-21.49-48-48 21.49-48 48-48 48 21.49 48 48zm-48 368c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zm208-208c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zM96 256c0-26.51-21.49-48-48-48S0 229.49 0 256s21.49 48 48 48 48-21.49 48-48zm12.922 99.078c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48c0-26.509-21.491-48-48-48zm294.156 0c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48c0-26.509-21.49-48-48-48zM108.922 60.922c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.491-48-48-48z"
    />
  </svg>
</span>
`;
